packages/search-input/src/SearchInput/SearchInput.spec.tsx

```diff
@@ -0,0 +1,436 @@
+import React from 'react';
+import {
+  createEvent,
+  fireEvent,
+  waitForElementToBeRemoved,
+} from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
+import { axe } from 'jest-axe';
+
+import * as _LGTL from '@leafygreen-ui/testing-lib';
+
+import { renderSearchInput } from '../utils/SearchInput.testutils';
+import { SearchInput, SearchResult, SizeVariant } from '..';
+
+const resultClickHandler = jest.fn();
+
+const defaultProps = {
+  className: 'test-text-input-class',
+  placeholder: 'This is some placeholder text',
+  onChange: jest.fn(),
+  onClear: jest.fn(),
+  onBlur: jest.fn(),
+  children: [
+    <SearchResult key="a" onClick={resultClickHandler}>
+      Apple
+    </SearchResult>,
+    <SearchResult key="b" onClick={resultClickHandler}>
+      Banana
+    </SearchResult>,
+    <SearchResult key="c" onClick={resultClickHandler}>
+      Carrot
+    </SearchResult>,
+    <SearchResult key="d" onClick={resultClickHandler}>
+      Dragonfruit
+    </SearchResult>,
+  ],
+};
+
+describe('packages/search-input', () => {
+  describe('a11y', () => {
+    test('does not have basic accessibility issues', async () => {
+      const { container } = renderSearchInput();
+      const results = await axe(container);
+      expect(results).toHaveNoViolations();
+    });
+  });
+
+  describe('Basic rendering', () => {
```

Comment:
We should add tests for the `state` prop, as well as `onChange` properly firing. I believe you test `onSubmit` below but that might be a different component's callback

Reply:
Added tests for `state`

I have tests for `onChange` when a character is typed and when the clear button is pressed. We also check that an Option's `onClick` handler is fired on enter key press